django/db/backends/mysql/features.py

```diff
@@ -104,6 +104,14 @@ def django_test_skips(self):
                 "update.tests.AdvancedTests.test_update_ordered_by_m2m_annotation",
                 "update.tests.AdvancedTests.test_update_ordered_by_m2m_annotation_desc",
             },
+            "MySQL does not support negative array indices in JSON paths": {
```

Comment:
I'd suggest adding a database feature flag since other databases may also have the limitation.